pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/BaseSingleTreeBuilder.java

```diff
@@ -133,7 +133,9 @@ static class Record {
           "Dimension: " + dimension + " does not have dictionary");
     }
 
-    TreeMap<AggregationFunctionColumnPair, AggregationSpec> aggregationSpecs = builderConfig.getAggregationSpecs();
+    TreeMap<AggregationFunctionColumnPair, AggregationSpec> aggregationSpecs =
+        StarTreeBuilderUtils.deduplicateAggregationSpecs(builderConfig.getAggregationSpecs());
```

Comment:
Similarly can we directly resolve the `AggregationFunctionType` within the builder config?

Reply:
This comment is discussed [here](https://github.com/apache/pinot/pull/12164#discussion_r1434289093).